### ISSUE TITLE
Warn when overriding a property with a method

### ISSUE THREAD

**Issue by Tadaboody** [PyCQA/pylint#2670]
### Additional context
For an example the error caused by this:
```py
import abc


class SuperClass(abc.ABC):
    @property
    @abc.abstractmethod
    def class_prop(self):
        """A type needed for the implementation"""


class NoProp(SuperClass):
    class A:
        def __init__(self, a=None):
            self.a = a or 0

        def method(self, param):
            return self.a + param

    def class_prop(self):
        return self.A


class Prop(SuperClass):
    class A:
        def __init__(self, a=None):
            self.a = a or 0

        def method(self, param):
            return self.a + param

    @property
    def class_prop(self):
        return self.A


for impl in (Prop, NoProp):
    impl().class_prop().method(0)
```
```sh
$ pylint pylint_example.py --disable=too-few-public-methods

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

$ python pylint_example.py
Traceback (most recent call last):
  File "pylint_example.py", line 37, in <module>
    impl().class_prop().method(0)
TypeError: method() missing 1 required positional argument: 'param'
```
The exception looks super weird because `param` is passed! but class_prop() is a type in one case and an instance in another

This could be prevented super early by emmiting a warning similar to (arguments-differ).

**For a real world example** see [the error](https://pastebin.com/7NiENEgN) and [the fix](https://github.com/Tadaboody/good_smell/pull/40/commits/76ec7fdbac27486aa78357716a1f254b472364a3)

**Comment by PCManticore** [comment]
This makes sense.
